leaders/src/main/java/edu/cornell/partitioner/ClassPartitioner.java

```diff
@@ -1,52 +1,102 @@
-     */
-    public @NonNull List<Set<String>> getClasses() {
-        return new ArrayList<>(classes);
-    }
+            // URL of the git repository
+            root.addProperty("url", System.getenv(Main.ENV_REPO_URL));
 
-    /**
-     * Partitions the test classes.
-     * @param tests the test classes to partition
-     * @param partitions the number of partitions
-     * @return the list of test partitions
-     */
-    private List<Set<String>> partitionClasses(List<String> tests, int partitions) {
-        List<Set<String>> classes = partitionMethod.partition(tests, partitions);
-        return classes;
+            // Branch of the git repository
+            root.addProperty("branch", System.getenv(Main.ENV_REPO_BRANCH));
+
+            // Number of workers
+            root.addProperty("num_workers",
+                    Integer.parseInt(System.getenv(Main.ENV_NUM_WORKERS)));
+
+            JsonArray testClasses = new JsonArray();
+            for (String test : testsList) {
+                JsonObject testClass = new JsonObject();
+                testClass.addProperty("name", test);
+                testClasses.add(testClass);
+            }
+            root.add("testclasses", testClasses);
+
+            // Convert JsonObject to String for the POST request body
+            String jsonBody = gson.toJson(root);
+
+            // Build HttpRequest with JSON body
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(jsonBody))
+                    .build();
+
+            // Send HttpRequest and receive HttpResponse
+            HttpResponse<String> response = client.send(request,
+                    HttpResponse.BodyHandlers.ofString());
+
+            // Parse JSON response to extract partitions
+            JsonObject jsonResponse = gson.fromJson(response.body(), JsonObject.class);
+            JsonArray partitions = jsonResponse.getAsJsonArray("partitions");
+            List<Set<String>> partitionsList = new ArrayList<>();
+
+            for (int i = 0; i < partitions.size(); i++) {
+                JsonObject partition = partitions.get(i).getAsJsonObject();
+                JsonArray testClassesArray = partition.getAsJsonArray("testclasses");
+                Set<String> testSet = new HashSet<>();
+
+                for (int j = 0; j < testClassesArray.size(); j++) {
+                    JsonObject testClassObject = testClassesArray.get(j).getAsJsonObject();
+                    testSet.add(testClassObject.get("name").getAsString());
+                }
+
+                partitionsList.add(testSet);
+            }
+
+            return partitionsList;
+
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error(e.toString());
+            return null; // or handle error appropriately
```

Comment:
I'm thinking we might wait to propagate this exception instead of catching it, logging it, and returning null. We still want the client of this method to handle an exception, so it might just be easier to declare that we throw `IOException` and `InterruptedException`.